packages/samples/atomic-react/src/App.tsx

```diff
@@ -1,218 +1,40 @@
-              template={MyTemplate}
-            />
-
-            <AtomicQueryError />
-            <AtomicNoResults />
-          </AtomicLayoutSection>
-          <AtomicLayoutSection section="pagination">
-            <AtomicLoadMoreResults />
-          </AtomicLayoutSection>
-        </AtomicLayoutSection>
-      </AtomicSearchLayout>
-    </AtomicSearchInterface>
-  );
-};
-
-function MyTemplate(result: Result) {
+  const [page, setPage] = useState(LIST_PAGE);
   return (
     <>
-      <AtomicResultSectionBadges>
-        <AtomicResultBadge field="ec_brand" />
-      </AtomicResultSectionBadges>
-      <AtomicResultSectionVisual>
-        <AtomicResultImage field="ec_images" />
-      </AtomicResultSectionVisual>
-      <AtomicResultSectionTitle>
-        <AtomicResultLink />
-      </AtomicResultSectionTitle>
-      <AtomicResultSectionTitleMetadata>
-        <AtomicResultRating field="ec_rating" />
-        <AtomicResultPrintableUri maxNumberOfParts={3} />
-      </AtomicResultSectionTitleMetadata>
-      <AtomicResultSectionEmphasized>
-        <AtomicResultNumber field="ec_price">
-          <AtomicFormatCurrency currency="USD" />
-        </AtomicResultNumber>
-      </AtomicResultSectionEmphasized>
-      <AtomicResultSectionExcerpt>
-        <AtomicResultText field="ec_shortdesc" />
-      </AtomicResultSectionExcerpt>
-      <AtomicResultSectionBottomMetadata>
-        <AtomicResultFieldsList>
-          {result.raw.cat_platform !== undefined && (
-            <>
-              <span className="field-label">
-                <AtomicText value="Platform" />
-              </span>
-              <AtomicResultText field="cat_platform" />
-            </>
-          )}
-          {result.raw.cat_condition !== undefined && (
-            <>
-              <span className="field-label">
-                <AtomicText value="Condition" />
-              </span>
-              <AtomicResultText field="cat_condition" />
-            </>
-          )}
-          {result.raw.cat_categories !== undefined && (
-            <>
-              <span className="field-label">
-                <AtomicText value="Tags" />
-              </span>
-              <AtomicResultMultiValueText field="cat_categories" />
-            </>
-          )}
-        </AtomicResultFieldsList>
-      </AtomicResultSectionBottomMetadata>
+      <header>
+        <span className="pageTitle">{page} example</span>
+        <ul>
+          <HeaderLink page={LIST_PAGE} currentPage={page} setPage={setPage} />
+          <HeaderLink
+            page={FOLDED_LIST_PAGE}
+            currentPage={page}
+            setPage={setPage}
+          />
+          <HeaderLink
+            page={FOLDED_CUSTOM_CHILDREN_PAGE}
+            currentPage={page}
+            setPage={setPage}
+          />
+        </ul>
+      </header>
+      {page === LIST_PAGE && <List />}
+      {page === FOLDED_LIST_PAGE && <Folded />}
+      {page === FOLDED_CUSTOM_CHILDREN_PAGE && <CustomChildren />}
```

Comment:
Is the main difference between these pages the result list? If so having multiple fully fledged apps makes it harder to maintain VS having something like

```
<AtomicApp>
   <AtomicFoldedResultList ...  />
 </AtomicApp>
```

Which would render the slotted result list component in the right spot of the app.

Reply:
yeah, I actually also changed the engine, but you're right. let me suggest something